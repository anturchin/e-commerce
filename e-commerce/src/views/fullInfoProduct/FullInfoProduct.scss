.product__info {
    display: flex;
    flex-direction: row-reverse;
    justify-content: center;
    align-items: center;
    gap: 15px;

    padding: 25px;
}
