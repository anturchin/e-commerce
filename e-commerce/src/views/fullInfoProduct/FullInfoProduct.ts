import { View } from '../View';
import { DescriptionProduct } from './descriptionProduct/DescriptionProduct';
import { NameProduct } from './nameProduct/NameProduct';
// import { ImgProduct } from './imgProduct/ImgProduct';
import { ContainerImg } from './containerForImg/ContainerForImg';
import { ContainerColumn } from './container/Container';
import { PriceProduct } from './priceProduct/PriceProduct';
import { SaleProduct } from './saleProduct/SaleProduct';
import { SlidesShow } from '../slidesShow/SlidesShow';

import './FullInfoProduct.scss';
import { ModalWin } from './modalWin/ModalWin';

export class FullInfoProduct extends View {
    private modal: ModalWin;

    constructor(
        productTitle: string,
        productImg: string[],
        productDescription: string,
        productPrice: string,
        productSale: string
    ) {
        super({ tag: 'div', classNames: ['product__info'] });
        this.modal = new ModalWin(productImg, 0);
        this.viewHtmlElement.addInnerElement(this.modal.getElement());
        this.setupProductInfo(
            productTitle,
            productImg,
            productDescription,
            productPrice,
            productSale
        );
    }

    private setupProductInfo(
        productTitle: string,
        productImg: string[],
        productDescription: string,
        productPrice: string,
        productSale: string
    ): void {
        const containerInfo = new ContainerColumn().getElement();

        const nameProduct = new NameProduct(productTitle).getElement();
        const descriptionProduct = new DescriptionProduct(productDescription).getElement();
        const price = new PriceProduct(productPrice).getElement();
        const sale = new SaleProduct(productSale).getElement();

        containerInfo.append(nameProduct, descriptionProduct, price, sale);
        this.viewHtmlElement.addInnerElement(containerInfo);

        const containerImg = new ContainerImg().getElement();
        const slidesShow = new SlidesShow(productImg).getElement();
        // slidesShow.classList.add('my-slides');
        containerImg.append(slidesShow);
        this.viewHtmlElement.addInnerElement(containerImg);

        slidesShow.querySelectorAll('img').forEach((img, index) => {
            img.addEventListener('click', () => {
                this.modal.setModalWin(productImg, index);
                this.modal.showModal();
            });
        });
    }
}
