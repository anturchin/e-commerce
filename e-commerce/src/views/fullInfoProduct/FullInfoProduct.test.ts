import { describe, expect, test } from '@jest/globals';
import { FullInfoProduct } from './FullInfoProduct';

describe('FullInfoProduct', () => {
    test('should match', () => {
        const fullInfoProduct = new FullInfoProduct(
            'productTitle',
            [''],
            'productDescription',
            'productPrice',
            'productSale'
        );
        const renderedFullInfoProduct = fullInfoProduct.getElement();
        expect(renderedFullInfoProduct.outerHTML).toMatchSnapshot();
    });
});
