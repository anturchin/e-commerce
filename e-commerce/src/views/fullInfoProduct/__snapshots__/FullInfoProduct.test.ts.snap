// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullInfoProduct should match 1`] = `"<div class="product__info"><div class="modal-win" style="display: none;"><button class="button">Close</button><div class="mySlides fade"><div class="slides" style="display: block;"><img class="img-slide modal-slide" src=""></div><button class="prev">prev</button><button class="next">next</button></div></div><div class="conteinerInfo"><h4 class="product__title">productTitle</h4><p class="product__description">productDescription</p><p class="product__price">productPrice</p><p class="product__sale-price">productSale</p></div><div class="conteiner__img"><div class="mySlides fade"><div class="slides" style="display: block;"><img class="img-slide modal-slide" src=""></div><button class="prev">prev</button><button class="next">next</button></div></div></div>"`;
